dbms/programs/server/MySQLHandler.cpp

```diff
@@ -28,24 +30,33 @@ namespace DB
 using namespace MySQLProtocol;
 
 
+#if USE_POCO_NETSSL
 using Poco::Net::SecureStreamSocket;
 using Poco::Net::SSLManager;
+#endif
 
 
 namespace ErrorCodes
 {
     extern const int MYSQL_CLIENT_INSUFFICIENT_CAPABILITIES;
     extern const int OPENSSL_ERROR;
+    extern const int SUPPORT_IS_DISABLED;
 }
 
-MySQLHandler::MySQLHandler(IServer & server_, const Poco::Net::StreamSocket & socket_, RSA & public_key_, RSA & private_key_, bool ssl_enabled, size_t connection_id_)
+MySQLHandler::MySQLHandler(IServer & server_, const Poco::Net::StreamSocket & socket_,
+#if USE_SSL
```

Comment:
It's a bad design to have a different interface depending on macro - better to have derived class.
Same for every occurrence below.